src/main/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/MTAmazonDynamoDBBySharedTable.java

```diff
@@ -219,74 +233,69 @@ public String toString() {
-//            this.tenant = tenant;
-//            this.tableName = tableName;
-//            this.processor = processor;
-//        }
-//        @Override
-//        public void initialize(InitializationInput initializationInput) {
-//            processor.initialize(initializationInput);
-//        }
-//
-//        @Override
-//        public void processRecords(ProcessRecordsInput processRecordsInput) {
-//            List<com.amazonaws.services.kinesis.model.Record> records = processRecordsInput.getRecords().stream()
-//                    .map(RecordAdapter.class::cast).map(this::toMTRecord).collect(toList());
-//            processor.processRecords(processRecordsInput.withRecords(records));
-//        }
-//
-//        private com.amazonaws.services.kinesis.model.Record toMTRecord(RecordAdapter adapter) {
-//            Record r = adapter.getInternalObject();
-//            return new RecordAdapter(new MTRecord() //
-//                    .withAwsRegion(r.getAwsRegion()) //
-//                    .withDynamodb(r.getDynamodb()) //
-//                    .withEventID(r.getEventID()) //
-//                    .withEventName(r.getEventName()) //
-//                    .withEventSource(r.getEventSource()) //
-//                    .withEventVersion(r.getEventVersion()) //
-//                    .withContext(tenant) //
-//                    .withTableName(tableName));
-//        }
-//
-//        @Override
-//        public void shutdown(ShutdownInput shutdownInput) {
-//            processor.shutdown(shutdownInput);
-//        }
-//
-//    }
+    private class RecordProcessor implements IRecordProcessor {
+        private final IRecordProcessor processor;
+        private final DynamoTableDescription physicalTable;
+
+        RecordProcessor(IRecordProcessor processor, DynamoTableDescription physicalTable) {
+            this.processor = processor;
+            this.physicalTable = physicalTable;
+        }
+
+        @Override
+        public void initialize(InitializationInput initializationInput) {
+            processor.initialize(initializationInput);
+        }
+
+        @Override
+        public void processRecords(ProcessRecordsInput processRecordsInput) {
+            List<com.amazonaws.services.kinesis.model.Record> records = processRecordsInput.getRecords().stream()
+                    .map(RecordAdapter.class::cast).map(this::toMTRecord).collect(toList());
+            processor.processRecords(processRecordsInput.withRecords(records));
+        }
+
+        private com.amazonaws.services.kinesis.model.Record toMTRecord(RecordAdapter adapter) {
+            Record r = adapter.getInternalObject();
+            StreamRecord streamRecord = r.getDynamodb();
+            FieldValue fieldValue = new FieldPrefixFunction(".").reverse(streamRecord.getKeys().get(physicalTable.getPrimaryKey().getHashKey()).getS());
+            getMTContext().setContext(fieldValue.getMtContext());
```

Comment:
why set mt context here?

Reply:
Unset context in a finally block ... 101ee1ad62730a68fb3999ce33fe5782e3c072db

FWIW, also fixed some formatting ... f2da4a6fda9e241c1963f8d1710e86753e338f1c